mlfinlab/structural_breaks/sadf.py

```diff
@@ -130,27 +134,30 @@ def _get_betas(X: pd.DataFrame, y: pd.DataFrame) -> Tuple[np.array, np.array]:
     return b_mean, b_var
 
 
-def _sadf_outer_loop(X: pd.DataFrame, y: pd.DataFrame, min_length: int, molecule: list) -> pd.Series:
+def _sadf_outer_loop(X: pd.DataFrame, y: pd.DataFrame, min_length: int, model: str, phi: float,
+                     molecule: list) -> pd.Series:
     """
     This function gets SADF for t times from molecule
 
     :param X: (pd.DataFrame) of features(factors)
     :param y: (pd.DataFrame) of outcomes
     :param min_length: (int) minimum number of observations
+    :param model: (str) either 'linear', 'quadratic', 'sm_poly_1', 'sm_poly_2', 'sm_exp', 'sm_power'
+    :param phi: (float) in [0, 1], penalize large sample lengths when computing SMT
     :param molecule: (list) of indices to get SADF
     :return: (pd.Series) of SADF statistics
     """
     sadf_series = pd.Series(index=molecule)
     for index in molecule:
         X_subset = X.loc[:index].values
         y_subset = y.loc[:index].values.reshape(-1, 1)
-        value = _get_sadf_at_t(X_subset, y_subset, min_length)
+        value = _get_sadf_at_t(X_subset, y_subset, min_length, model, phi)
         sadf_series[index] = value
     return sadf_series
 
 
 def get_sadf(series: pd.Series, model: str, lags: Union[int, list], min_length: int, add_const: bool = False,
-             num_threads: int = 8) -> pd.Series:
+             phi: float = 0, num_threads: int = 8) -> pd.Series:
     """
     Multithread implementation of SADF, p. 258-259
```

Comment:
This may be the best place to extend the dosctring to let the user know what your additional implementation does as well.

Reply:
Added docstring